src/ibl_third_party_auth/views.py

```diff
@@ -15,3 +37,128 @@
 def back_channel_logout(request, backend):
     """Back Channel logout"""
     return backchannel_logout.back_channel_logout(request, backend)
+
+
+@method_decorator(csrf_exempt, name="dispatch")
+class DMTokenView(OAuthLibMixin, View):
+    server_class = OAuth2Server
+    validator_class = OAuth2Validator
+
+    oauthlib_backend_class = OAuthLibCore
+
+    @staticmethod
+    def parse_iso8601_to_utc(iso_string: str) -> datetime.datetime:
+        """
+        Parses an ISO 8601 formatted string into a timezone-aware UTC datetime object.
+
+        - If the string contains timezone information (e.g., +02:00, -05:00, Z),
+          it's converted to UTC.
+        - If the string does *not* contain timezone information (naive datetime),
+          it's assumed to represent a UTC time, and UTC timezone info is attached.
+
+        Args:
+            iso_string: The string formatted according to ISO 8601.
+
+        Returns:
+            A timezone-aware datetime object representing the time in UTC.
+
+        Raises:
+            ValueError: If the string is not a valid ISO 8601 format
+                        parsable by datetime.fromisoformat.
+        """
+        # Python < 3.11 doesn't handle 'Z' directly in fromisoformat
+        # We replace 'Z' with '+00:00' for compatibility
+        if iso_string.endswith("Z"):
+            iso_string = iso_string[:-1] + "+00:00"
+        try:
+            dt = datetime.datetime.fromisoformat(iso_string)
+            if dt.tzinfo is None or dt.tzinfo.utcoffset(dt) is None:
+                dt_utc = dt.replace(tzinfo=datetime.timezone.utc)
+            else:
+                dt_utc = dt.astimezone(datetime.timezone.utc)
+            return dt_utc
+
+        except ValueError as e:
+            raise ValueError(f"Invalid ISO 8601 string format: '{iso_string}'") from e
+
+    def post(self, request, *args, **kwargs):
+        url, headers, body, status = self.create_token_response(request)
+        if status == 200:
+            access_token = json.loads(body).get("access_token")
+            token = get_access_token_model().objects.get(token=access_token)
```

Comment:
We should do a `select_related("user")` so we're not doing an additional query to get the `user` object when we access `token.user`